packages/astro/src/core/build/plugins/plugin-content.ts

```diff
@@ -290,13 +355,65 @@ async function generateContentManifest(
 			);
 		}
 	}
+	
+	const [lockfiles, configs] = await Promise.all([
+		lockfilesHash(opts.settings.config.root),
+		configHash(opts.settings.config.root)
+	]);
+
+	manifest.lockfiles = lockfiles;
+	manifest.configs = configs;
 
 	await Promise.all(promises);
 	return manifest;
 }
 
-function checksum(data: string): string {
-	return createHash('sha1').update(data).digest('base64');
+async function pushBufferInto(fileURL: URL, buffers: Uint8Array[]) {
+	try {
+		const handle = await fsMod.promises.open(fileURL, 'r');
+		const data = await handle.readFile();
+		buffers.push(data);
+		await handle.close();
+	} catch {
+		// File doesn't exist, ignore
+	}
+}
+
+async function lockfilesHash(root: URL) {
+	// Order is important so don't change this.
+	const lockfiles = ['package-lock.json', 'pnpm-lock.yaml', 'yarn.lock', 'bun.lockb'];
+	const datas: Uint8Array[] = [];
+	const promises: Promise<void>[] = [];
+	for(const lockfileName of lockfiles) {
+		const fileURL = new URL(encodeURI(joinPaths(root.toString(), lockfileName)));
+		promises.push(pushBufferInto(fileURL, datas));
+	}
+	await Promise.all(promises);
+	return checksum(...datas);
+}
+
+async function configHash(root: URL) {
+	const configFileNames = configPaths;
+	for(const configPath of configFileNames) {
+		try {
+			const fileURL = new URL(encodeURI(joinPaths(root.toString(), configPath)));
+			const handle = await fsMod.promises.open(fileURL, 'r');
+			const data = await handle.readFile();
+			const hash = checksum(data);
+			await handle.close();
```

Comment:
This code looks the same as `pushBufferInto()`, should we use it instead?

Also, why not use `await fsMod.promises.readFile(fileURL)` instead? It returns a buffer by default